sdk/src/client/api/block_builder/input_selection/requirement/amount.rs

```diff
@@ -13,18 +13,22 @@ use crate::{
             unlock_condition::StorageDepositReturnUnlockCondition, AccountOutputBuilder, AccountTransition,
             FoundryOutputBuilder, NftOutputBuilder, Output, OutputId, Rent,
         },
+        slot::SlotIndex,
     },
 };
 
 /// Get the `StorageDepositReturnUnlockCondition`, if not expired.
-pub(crate) fn sdruc_not_expired(output: &Output, current_time: u32) -> Option<&StorageDepositReturnUnlockCondition> {
+pub(crate) fn sdruc_not_expired(
+    output: &Output,
+    slot_index: SlotIndex,
+) -> Option<&StorageDepositReturnUnlockCondition> {
     // PANIC: safe to unwrap as outputs without unlock conditions have been filtered out already.
     let unlock_conditions = output.unlock_conditions().unwrap();
 
     unlock_conditions.storage_deposit_return().and_then(|sdr| {
         let expired = unlock_conditions
             .expiration()
-            .map_or(false, |expiration| current_time >= expiration.timestamp());
+            .map_or(false, |expiration| slot_index >= expiration.slot_index());
```

Comment:
That's not that simple anymore https://github.com/iotaledger/tips-draft/blob/tip38/tips/TIP-0038/tip-0038.md#additional-semantic-transaction-validation-rules-2

Reply:
Fine for me